floyd/client/files.py

```diff
@@ -114,9 +119,35 @@ def sizeof_fmt(num, suffix='B'):
     return "%.1f%s%s" % (num, 'Yi', suffix)
 
 
+def warn_purge_exit(info_msg, filename, exit_msg):
+    """
+    Warn the user that's something went wrong,
+    remove the tarball and provide an exit message.
+    """
+    floyd_logger.info(info_msg)
+    rmtree(os.path.dirname(filename))
+    sys.exit(exit_msg)
+
+
 def create_tarfile(source_dir, filename="/tmp/contents.tar.gz"):
     """
     Create a tar file with the contents of the current directory
     """
-    with tarfile.open(filename, "w:gz") as tar:
-        tar.add(source_dir, arcname=os.path.basename(source_dir))
+    try:
+        # Define the default signal handler for catching: Ctrl-C
+        signal.signal(signal.SIGINT, signal.default_int_handler)
+        with tarfile.open(filename, "w:gz") as tar:
+            tar.add(source_dir, arcname=os.path.basename(source_dir))
+
+    except OSError as e:
+        # OSError: [Errno 13] Permission denied
+        if e.errno == errno.EACCES:
+            warn_purge_exit(info_msg="Permission denied. Removing compressed data...",
+                            filename=filename,
+                            exit_msg=("Permission denied. Make sure to have read permission "
+                                      "for all the files and directories in the path."))
```

Comment:
`in the path` should be `in the current directory`?

Reply:
@narenst

Reply:
Since this is a generic function, the source_dir might not be current directory where the command is run from. So I suggested "path". I'm fine with "current directory" as well.

Reply:
@narenst after your explanation, I think we should use your suggestion and print also the `source_dir`